pints/functionaltests/functest_haario_bardenet_acmc.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+
+from __future__ import division
+
+import numpy as np
+
+import pints
+import pints.toy
+
+
+class TestHaarioBardenetACMCOn2dimGaussianDistribution(object):
+    """
+    Tests the pints.HaarioBardenetACMC on a standard 2 dimensional Gaussian
+    distribution.
+    """
+    def __init__(self):
```

Comment:
Not sure why this is 4 methods?

Would it make sense to have a single method that returns `name, results` ? Then we could have some class for the TestHaarioBardenetACMC where each method is a different test?